src/OpenTelemetry.Contrib.PersistentStorage/LocalFileStorage.cs

```diff
@@ -0,0 +1,177 @@
+            GC.SuppressFinalize(this);
+        }
+
+        public void Dispose(bool disposing)
+        {
+            if (!this.disposedValue)
+            {
+                if (disposing)
+                {
+                    this.maintenanceTimer.Dispose();
+                }
+
+                this.disposedValue = true;
+            }
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<IPersistentBlob> GetBlobs()
+        {
+            var retentionDeadline = DateTime.UtcNow - TimeSpan.FromMilliseconds(this.retentionPeriodInMilliseconds);
+
+            foreach (var file in Directory.EnumerateFiles(this.directoryPath, "*.blob", SearchOption.TopDirectoryOnly).OrderByDescending(f => f))
+            {
+                DateTime fileDateTime = PersistentStorageHelper.GetDateTimeFromBlobName(file);
+                if (fileDateTime > retentionDeadline)
+                {
+                    yield return new LocalFileBlob(file);
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        public IPersistentBlob GetBlob()
+        {
+            return this.GetBlobs().FirstOrDefault();
+        }
+
+        /// <inheritdoc/>
+        public IPersistentBlob CreateBlob(byte[] buffer, int leasePeriodMilliseconds = 0)
+        {
+            if (!this.CheckStorageSize())
+            {
+                return null;
+            }
+
+            try
+            {
+                var blobFilePath = Path.Combine(this.directoryPath, PersistentStorageHelper.GetUniqueFileName(".blob"));
+                var blob = new LocalFileBlob(blobFilePath);
+                return blob.Write(buffer, leasePeriodMilliseconds);
+            }
+            catch (Exception ex)
+            {
+                StorageEventSource.Log.Warning("CreateBlob has failed.", ex);
+                return null;
+            }
+        }
+
+        private void OnMaintenanceEvent(object source, ElapsedEventArgs e)
+        {
+            try
+            {
+                if (!Directory.Exists(this.directoryPath))
+                {
+                    Directory.CreateDirectory(this.directoryPath);
+                }
+            }
+            catch (Exception ex)
+            {
+                StorageEventSource.Log.Error($"Error creating directory {this.directoryPath}", ex);
+                return;
+            }
+
+            PersistentStorageHelper.RemoveExpiredBlobs(this.directoryPath, this.retentionPeriodInMilliseconds, this.writeTimeoutInMilliseconds);
+        }
+
+        private bool CheckStorageSize()
+        {
+            var size = PersistentStorageHelper.GetDirectorySize();
+            if (size >= this.maxSizeInBytes)
+            {
+                StorageEventSource.Log.Warning($"Persistent storage max capacity has been reached. Currently at {size / 1024} KB. " +
+                                                "Telemetry will be lost. Please consider increasing the value of storage max size in exporter config.");
```

Comment:
I suggest that we remove "Telemetry will be lost" here. Whether we drop telemetry or not is depending on multiple factors.